client/lib/fifo/fifo_unix.go

```diff
@@ -3,23 +3,34 @@
 package fifo
 
 import (
-	"context"
+	"fmt"
 	"io"
 	"os"
-	"syscall"
 
-	cfifo "github.com/containerd/fifo"
+	"golang.org/x/sys/unix"
 )
 
-// New creates a fifo at the given path and returns an io.ReadWriteCloser for it
-// The fifo must not already exist
-func New(path string) (io.ReadWriteCloser, error) {
-	return cfifo.OpenFifo(context.Background(), path, syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0600)
+// New creates a fifo at the given path, and returns an open function for reading.
+// The fifo must not exist already, or that it's already a fifo file
+//
+// It returns a reader open function that may block until a writer opens
+// so it's advised to run it in a goroutine different from reader goroutine
+func New(path string) (func() (io.ReadCloser, error), error) {
```

Comment:
Nit - Could we name these two methods `OpenForRead` and `OpenForWrite` instead or `New` and `Open`, for clarity? Its a bit strange that open is for writes and new is for reads.

Reply:
I'll defer to @nickethier or @schmichael as the names predated this PR.

Reply:
+1 to renaming.